tests/run_algorithm_test.py

```diff
@@ -24,12 +26,41 @@ def get_train_test() -> Tuple[DataTuple, DataTuple]:
     return train_test
 
 
+def count_true(mask: 'np.ndarray[bool]') -> int:
+    """Count the number of elements that are True"""
+    return mask.nonzero()[0].shape[0]
+
+
 def test_can_load_test_data():
     train, test = get_train_test()
     assert train is not None
     assert test is not None
 
 
+def test_run_parallel():
+    data = get_train_test()
+    result = run_in_parallel([LR(), SVM(), Majority()], data)
+    assert count_true(result[0].values == 1) == 211
+    assert count_true(result[0].values == -1) == 189
+    assert count_true(result[1].values == 1) == 201
+    assert count_true(result[1].values == -1) == 199
+    assert count_true(result[2].values == 1) == 0
+    assert count_true(result[2].values == -1) == 400
+
+
+def test_run_parallel_multi_data():
+    data0 = get_train_test()
+    data1 = get_train_test()
+    data2 = get_train_test()
+    result = run_in_parallel([LR(), SVM(), Majority()], [data0, data1, data2])
```

Comment:
It would be really nice if there were a 3rd case, where 
`result = run_in_parallel(model(), [data1, data2, data3])`
would apply the one model to multiple datasets. The use case would be where we have created multiple variations on a dataset and we want to run one model over it (as we did recently)